index.html

```diff
@@ -1884,50 +1909,72 @@ <h2>The <dfn>NDEFRecordType</dfn> string</h2>
       <td><dfn>"`empty`"</dfn></td>
       <td><i>unused</i></td>
       <td><i>unused</i></td>
-      <td><a>Empty record</a></td>
+      <td><a>Empty record</a> (TNF=0)</td>
     </tr>
     <tr>
       <td><dfn>"`text`"</dfn></td>
       <td><i>unused</i></td>
       <td>{{DOMString}}</td>
-      <td><a>Well-known type</a> record with type "`T`"</td>
+      <td><a>Well-known type</a> (TNF=1) record with TYPE="`T`"</td>
     </tr>
     <tr>
       <td><dfn>"`url`"</dfn></td>
       <td><i>unused</i></td>
       <td>{{DOMString}}</td>
-      <td><a>Well-known type</a> record with type "`U`"</td>
+      <td><a>Well-known type</a> (TNF=1) record with TYPE="`U`"</td>
     </tr>
     <tr>
       <td><dfn>"`smart-poster`"</dfn></td>
       <td><i>unused</i></td>
       <td>{{NDEFMessage}}</td>
-      <td><a>Well-known type</a> record with type "`Sp`"</td>
+      <td><a>Well-known type</a> (TNF=1) record with TYPE="`Sp`"</td>
+    </tr>
+    <tr>
+      <td><dfn>"`absolute-url`"</dfn></td>
+      <td><i>unused</i></td>
+      <td>{{DOMString}} <i>url</i></td>
+      <td><a>Absolute-URL record</a> (TNF=2) with TYPE=<i>url</i></td>
     </tr>
     <tr>
       <td><dfn>"`json`"</dfn></td>
       <td><a>JSON MIME type</a></td>
       <td>
         [= JSON type =]
       </td>
-      <td><a>MIME type record</a> with type equal to
-        <a>MIME type</a>.
+      <td><a>MIME type record</a> (TNF=2) with TYPE= [= JSON type =].
       </td>
     </tr>
     <tr>
       <td><dfn>"`opaque`"</dfn></td>
-      <td><a>MIME type</a></td>
+      <td>[= MIME type =]</a></td>
       <td>{{ArrayBuffer}}</td>
-      <td><a>MIME type record</a></td>
+      <td><a>MIME type record</a> (TNF=3) with TYPE= [= MIME type =]</td>
     </tr>
     <tr>
-      <td><a>external type</a></td>
+      <td><i>[= external type =]</i></td>
+      <td><i>unused</i></td>
+      <td>
+        {{ArrayBuffer}} or any<br>
+        <a>typed array type</a> or<br>
+        {{NDEFMessage}}
```

Comment:
As we are trying to align our APIs with the NFC forum ones, and we are sort of sure that a smart poster may have a ndef message as payload, but I'm wondering whether it's also true for an external record? I'm not familiar with nfc forum regulations about that.

Reply:
Payload of external type is up to the applications to handle (since they also define the type).
Web NFC may play nice and add this convenience feature for apps to try parsing the payload as NDEFMessage and make it easy to embed an NDEFMessage into an external type record.

The other take would be to only allow ArrayBuffer (which is closer to the NFC specs). One could argue that if developers use external types, they know well enough what they are doing.

@kenchris what's your take?

Reply:
Web NFC should definitely add convenience to write and read custom embedded records - very few native APIs do that and I think that has been holding NFC back somewhat. It is pretty clear from the NDEF specs that this is an intended use-case and it is standardized how it works. NDEF specs don't talk about user facing APIs, just how things are stored and read

Reply:
> One could argue that if developers use external types, they know well enough what they are doing.

Such an argument is what limits adoption. There are quite some details to get right for average joe developer and people will just not do it

Reply:
OK, we stay with the swiss army knife :).